Dockerfile

```diff
@@ -1,21 +1,40 @@
+################################
+# Build stage for the frontend #
+################################
+FROM node:12 as BUILDER
+
+WORKDIR /usr/src/client
+# Copy first package.json so changes in code dont require to reinstall all npm modules
+COPY client/package.json client/package-lock.json ./
+RUN npm i
+
+COPY . /usr/src
```

Comment:
It might be better to limit this to the `client` and `server` folders (and in the future exclude the client/tests folder as well).

Reply:
It is already added to the .dockerignore. If new folders are added in the future that needs to not be copied, they will need to be added to the  .dockerignore
You can see what actually gets added by running `docker run --rm -it $(docker build -q .) bash` and then inspecting the contents.
From my run, after merging all open PR:
`PA.spec  __init__.py  api     auth.py	 data	 planarserver.py   routes.py  server_config.cfg  t.planar.sqlite  utils.py VERSION  __pycache__  app.py  config.py  models  requirements.txt  save.py    static		 templates`

In any case, the only image that is distributed is the last one. The image that compiles the frontend is a bit more than 1GB in size at the end (node is heavy), but as it only ever lives in developer machines, its not that bad. To reduce that size, the base image could be changed to 12-alpine, which [has problems with some libraries](https://hub.docker.com/_/node/), but seems to work for this project (docker recognizes the built static files as the same so the final image has the same checksum). It reduces the build image size to 340MB. 

We could also not copy the full server folder to the build images, as it is not strictly needed, but copying it all is much simpler.

Reply:
Right, yeah my bad, thanks for clearing that up.